api/src/main/java/org/apache/iceberg/Scan.java

```diff
@@ -114,4 +133,19 @@
    * @return an Iterable of tasks for this scan
    */
   CloseableIterable<CombinedScanTask> planTasks();
+
+  /**
+   * Returns the target split size for this scan.
+   */
+  long targetSplitSize();
```

Comment:
I moved those 3 split methods so that we can share the implementations in `BaseScan` class. 

Alternatively, we can move them into a static util class. Would love to get some feedback here.
```
public class ScanUtil {
  public static long targetSplitSize(TableOperations ops, TableScanContext context) {
    long tableValue = ops.current().propertyAsLong(
        TableProperties.SPLIT_SIZE,
        TableProperties.SPLIT_SIZE_DEFAULT);
    return PropertyUtil.propertyAsLong(context.options(), TableProperties.SPLIT_SIZE, tableValue);
  }
}
```